obj/gebaeude.cc

```diff
@@ -403,7 +404,13 @@ image_id gebaeude_t::get_image() const
 
 image_id gebaeude_t::get_outline_image() const
 {
-	if(env_t::hide_buildings!=0  &&  env_t::hide_with_transparency  &&  !zeige_baugrube) {
+	// get city highlight & city highlighted
+	city_info_t *w_city = dynamic_cast<city_info_t*>(win_get_magic((ptrdiff_t)this->get_stadt()));
+	bool city_highlight = w_city ? w_city->is_highlighted() : false;
+	stadt_t* city = w_city ? w_city->get_city() : nullptr;
```

Comment:
We can simply check this with `env_t::highlighted_city` instead of checking the state of the city info window.